CI broken: ThePebble missing IREE third_party/printf submodule

## Summary

All CI jobs on `main` (and every PR) are failing since #236 bumped IREE to `3.11.0rc20260311`. The `ThePebble.py` setup script does not fetch the `third_party/printf` submodule that IREE's runtime now depends on.

## Error

```
CMake Error at /github/home/.cache/ThePebble/iree/build_tools/cmake/external_cc_library.cmake:116 (target_sources):
  Cannot find source file:

    /github/home/.cache/ThePebble/iree/third_party/printf/src/printf/printf.c
```

## Cause

`ThePebble.py` line 90 hardcodes the IREE submodules to fetch:

```python
IREE_SUBMODULES = ["third_party/flatcc", "third_party/benchmark"]
```

IREE's `iree/base` runtime library now links `printf::printf`, defined in `build_tools/third_party/printf/CMakeLists.txt`, which references sources in the `third_party/printf` submodule. Since ThePebble doesn't fetch it, the cmake configure fails.

## Fix

Add `"third_party/printf"` to `IREE_SUBMODULES`:

```python
IREE_SUBMODULES = ["third_party/flatcc", "third_party/benchmark", "third_party/printf"]
```

## Affected

Every CI run since #236 merged (2026-03-10).

--- a/plugins/hipdnn-plugin/build_tools/ThePebble.py
+++ b/plugins/hipdnn-plugin/build_tools/ThePebble.py
@@ -87,7 +87,7 @@ THEROCK_DIR = PEBBLE_DIR / "TheRock"
 ROCM_LIBRARIES_REPO = "https://github.com/ROCm/rocm-libraries.git"
 IREE_REPO = "https://github.com/iree-org/iree.git"
 IREE_DIR = PEBBLE_DIR / "iree"
-IREE_SUBMODULES = ["third_party/flatcc", "third_party/benchmark"]
+IREE_SUBMODULES = ["third_party/flatcc", "third_party/benchmark", "third_party/printf"]
 HIPDNN_SRC_DIR = PEBBLE_DIR / "rocm-libraries"
 
 # ==============================================================================

